### PR TITLE
Add program-name into arguments of ytserver-all

### PR DESCRIPTION
**Pull request by koct9i**
Handle first argument of ytserver-all as target program name.

Example:
```
$ ./ytserver-all --help
Usage: ./ytserver-all [OPTIONS] [program-name] [ARG]...

Options:
  --svnrevision print svn version
  {-?|--help}   print usage
  --yt-version  print YT version and exit
  --version     print version and exit
  --yson        print build information in YSON
  --build       print build information and exit

Free args: min: 0, max: unlimited
  program-name  Program name to run

Programs:
  ytserver-master
  ytserver-clock
  ytserver-http-proxy
  ytserver-proxy
  ytserver-node
  ytserver-job-proxy
  ytserver-exec
  ytserver-tools
  ytserver-scheduler
  ytserver-controller-agent
  ytserver-log-tailer
  ytserver-discovery
  ytserver-timestamp-provider
  ytserver-master-cache
  ytserver-cell-balancer
  ytserver-queue-agent
  ytserver-tablet-balancer
  ytserver-cypress-proxy
  ytserver-query-tracker
  ytserver-tcp-proxy
  ytserver-replicated-table-tracker
```

### CODE REVIEW

File: yt/yt/server/all/main.cpp [hunk 1]
```diff
@@ -31,12 +31,24 @@ using namespace NYT;
 class TAllProgram
     : public TProgram
 {
+public:
+    TAllProgram()
+    {
+        // Fake option just to show in --help
```
Comment (gritukan) [inline yt/yt/server/all/main.cpp:37]: We end comments with a dot.

File: yt/yt/server/all/main.cpp [hunk 2]
```diff
@@ -31,12 +31,24 @@ using namespace NYT;
...
+    }
+
 private:
     void DoRun(const NLastGetopt::TOptsParseResult& /*result*/) override
     {
         Cerr << "Program " << Argv0_ << " is not known" << Endl;
         Exit(EProgramExitCode::OptionsError);
     }
+
+    TString ProgramName_;
```
Comment (gritukan) [inline yt/yt/server/all/main.cpp:51]: In class sections (public/private) fiends are declared before methods.

File: yt/yt/server/all/main.cpp [hunk 3]
```diff
@@ -49,6 +61,14 @@ void TryProgram(int argc, const char** argv, const TString& nameSuffix)
 
 int main(int argc, const char** argv)
 {
+    if (argc >= 2 &&
```
Comment (gritukan) [inline yt/yt/server/all/main.cpp:64]: Let's add a comment what is going on here.

File: yt/yt/server/all/main.cpp [hunk 4]
```diff
@@ -49,6 +61,15 @@ void TryProgram(int argc, const char** argv, const TString& nameSuffix)
 
 int main(int argc, const char** argv)
 {
+    // Shift arguments to handle "program-name" in first argument.
```
Comment (gritukan) [inline yt/yt/server/all/main.cpp:64]: I'd suggest to add `Allows to provide program name as an argument, for example, ./ytserver-all ytserver-clock`.

**Review by gritukan — APPROVED** [review]

**Review by chizhonkova — APPROVED** [review]

**Pull request closed without merging**